### PR TITLE
Add visitTransitiveScopeEndingUses and related cleanup

### PR DESCRIPTION
**Pull request by atrick**
In preparation for adding support for "reborrows" particularly in CanonicalizeOSSA.

It will be tested in the next PR. This just gets the CI pipeline rolling.

### CODE REVIEW

**Comment by atrick** [comment]
@swift-ci smoke test

File: lib/SIL/Utils/OwnershipUtils.cpp [hunk 1]
```diff
@@ -231,6 +231,26 @@ bool BorrowingOperand::visitLocalScopeEndingUses(
...
+  SmallSetVector<Operand *, 4> reborrows;
+  reborrows.insert(this->op);
+  auto visitEnd = [&](Operand *scopeEndingUse) {
+    if (scopeEndingUse->getOperandOwnership() == OperandOwnership::Reborrow) {
+      reborrows.insert(scopeEndingUse);
+      return true;
+    }
+    return func(scopeEndingUse);
+  };
+  for (unsigned idx = 0; idx < reborrows.size(); ++idx) {
```
Comment (gottesmm) [inline lib/SIL/Utils/OwnershipUtils.cpp:245]: Can you add a comment here explaining that reborrows is resized in the loop so we need to check the dynamic size here?

**Review by gottesmm — APPROVED** [review]
Just add a comment in a follow on.

**Comment by atrick** [comment]
Subsumed by https://github.com/apple/swift/pull/35671

**Pull request closed without merging**